chat_v2_message_counter_sample/src/main/java/com/zendesk/chat_v2/sample/UnreadMessageCounter.java

```diff
@@ -0,0 +1,73 @@
+package com.zendesk.chat_v2.sample;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import zendesk.chat.Chat;
+import zendesk.chat.ChatLog;
+import zendesk.chat.ChatState;
+import zendesk.chat.ObservationScope;
+import zendesk.chat.Observer;
+
+class UnreadMessageCounter {
+
+    private final UnreadMessageCounterListener unreadMessageCounterListener;
+
+    private boolean shouldCount;
+    private String lastChatLogId;
+    private String lastReadChatLogId;
+    private List<ChatLog> unreadChatLogs = new ArrayList<>();
+    private ObservationScope observationScope = new ObservationScope();
```

Comment:
I'm not sure we need this as a field now either if we are no longer cancelling :) I'd be inclined to say that we shouldn't worry about the cancellation for the purposes of this example so just inline the creation of the scope.